src/http-client/index.ts

```diff
@@ -14,9 +22,30 @@ abstract class HttpClient extends Dispatcher {
    */
   public static async request(options: HttpClientRequestOptions): Promise<Response> {
     let res = await this._request(options, false);
+    let txnAuthJSON: TxnAuthJSON | undefined;
+
     if (this.newTokenRequired(res, options.requiresNewToken)) {
       res = await this._request(options, true);
     }
+
+    if (options.txnAuth && options.txnAuth.init) {
```

Comment:
I figured this should be totally opt-in. Not a huge fan of the object with an `init` property, but with the need to pass in the options object specifically for this, I decided with the object approach.